pkg/v1/cli/artifact/interface.go

```diff
@@ -5,14 +5,31 @@
 // from different sources
 package artifact
 
+import (
+	"net/url"
+	"path/filepath"
+)
+
 // Artifact is an interface to download a single plugin binary.
 type Artifact interface {
 	// Fetch the binary for a plugin version.
 	Fetch() ([]byte, error)
 }
 
 // NewURIArtifact creates new artifacts based on the URI
-func NewURIArtifact(uri string) Artifact {
+func NewURIArtifact(uri string) (Artifact, error) {
+	u, err := url.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+
+	switch u.Scheme {
+	case uriSchemeGCP:
+		return NewGCPArtifact(u.Host, u.Path), nil
+	case uriSchemeLocal:
+		return NewLocalArtifact(filepath.Join(u.Host, u.Path)), nil
 	// TODO: Support other artifact types
-	return NewLocalArtifact(uri)
+	default:
+		return NewLocalArtifact(uri), nil
```

Comment:
nit: A comment might be useful on why we are defaulting to local artifacts. Something like, `Local artifacts might point to a relative path without specifying uriScheme as prefix hence defaulting to local artifacts if no uriScheme is provided.`